### PR TITLE
SW-2530 add correction factor limit and max intensity value to laserhead profile

### PR DESCRIPTION
**Pull request by JFSolution**
- add max_correction_factor and max_intensity_including_correction to laserhead profiles
- disable autoconnect to laser
- connect to laser after iobeam is connected and laserhead data is received

### CODE REVIEW

**Comment by JFSolution** [comment]
This PR currently also include SW-1989 as this introduced some changes I needed here as well

File: octoprint_mrbeam/__init__.py [hunk 1]
```diff
@@ -285,6 +292,45 @@ def initialize(self):
...
+
+        Returns:
+
+        """
+        self._logger.info("MrBeamPlugin on_laserhead_ready")
+        self._laserhead_ready = True
+        self._try_to_connect_laser()
+
+    def _try_to_connect_laser(self):
+        """
```
Comment (irlaec) [inline octoprint_mrbeam/__init__.py:324]:
```suggestion
        """```

File: octoprint_mrbeam/__init__.py [hunk 2]
```diff
@@ -285,6 +292,45 @@ def initialize(self):
...
+        Tries to connect the laser if both iobeam and laserhead are ready and the laser is not connected yet.
```
Comment (irlaec) [inline octoprint_mrbeam/__init__.py]: I think there's something wrong with the formatting here

File: tests/camera/test_camera.py [hunk 3]
```diff
@@ -19,23 +20,25 @@
...
-        cam.async_capture()
-        cam.wait()
-        assert worker.count == 2
-        assert cam.lastPic() is not None
+def test_normal_use(datafiles, mrbeam_plugin):
+    def mock_get(*args, **kwargs):
+        if args[0] == ["mrbeam", "mock", "img_static"]:
+            return str(datafiles / "raw.jpg")
+
+    with patch(
```
Comment (irlaec) [inline tests/camera/test_camera.py]: Maybe we could create a fixture with the MrbCamera (similar to what we do with mrbeam_plugin)?

**Review by irlaec — APPROVED** [review]

**Review by irlaec — APPROVED** [review]

File: tests/camera/test_camera.py (hunk 3 above)
Comment (JFSolution) [reply tests/camera/test_camera.py]: I was trying this but as I need to get the datafiles in there and these are parameterized, it was not possible

**Pull request merged**